hw/top_earlgrey/data/chip_testplan.hjson

```diff
@@ -3,18 +3,1146 @@
+            '''
+      milestone: V2
+      tests: []
+    }
+
+    // WATCHDOG (pre-verified IP) integration tests:
+    {
+      name: chip_wdog_bark
+      desc: '''Verify the watchdog bark reception in awake state.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: chip_wdog_bite
+      desc: '''Verify the watchdog bite causing reset in awake state.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: chip_sleep_wdog_bark_wake
+      desc: '''Verify the watchdog bark wake up from sleep state.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: chip_sleep_wdog_bite_wake
+      desc: '''Verify the watchdog bite reset from sleep state.
+            '''
+      milestone: V2
+      tests: []
+    }
+
+    // PLIC (pre-verified IP) integration tests:
+    {
+      name: chip_plic_all_irqs
+      desc: '''Verify all interrupts from all peripherals aggregated at the PLIC.
+
+            The automated SW test enables all interrupts at the PLIC to interrupt the core. It uses
+            the `intr_test` CSR in each peripheral to mock assert an interrupt, looping through all
+            available interrupts in that peripheral. The ISR verifies that the right interrupt
+            occurred. This is used as a catch-all interrupt test for all peripheral integration
+            testing within which functionally asserting an interrupt is hard to achieve or not of
+            high value.
+            '''
+      milestone: V2
+      tests: [""]
+    }
+    {
+      name: chip_plic_sw_irq
+      desc: '''Verify the SW interrupt to the CPU.
+
+            Enable all peripheral interrupts at PLIC. Enable all types of interrupt at the CPU core.
+            Write to the MSIP CSR to generate a SW interrupt to the CPU. Verify that the only
+            interrupt that is seen is the SW interrupt.
+            '''
+      milestone: V2
+      tests: [""]
+    }
+    {
+      name: chip_plic_nmi_irq
+      desc: '''Verify the NMI interrupt to the CPU and correctness of the cause.
+
+            TBD if multiple NMI irqs are OR-ed into the CPU (example - NMI from alert handler and
+            the watchdog bark), then map each test to this testpoint.
+            '''
+      milestone: V2
+      tests: [""]
+    }
+
+    // CLKMGR tests:
+    {
+      name: chip_sw_clk_off_trans
+      desc: '''Verify the ability to turn off the transactional clock via SW.
+
+             Ensure that activity in any of the IPs running on this clock prevents the clock from
+             actually being turned off. Verify that turning off this clock does not affect the other
+             derived clocks.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: chip_sw_clk_off_peri
+      desc: '''Verify the ability to turn off the peripheral clock via SW.
+
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: chip_clk_div
```

Comment:
@sriyerg did you say you would prefer this to be converted into assertions? If we do that, does that mean we won't need to have a test and can just rely on background checks?

Reply:
These are just test points in the testplan - these just indicate that "we need to absolutely verify this feature". How the feature actually gets verified is an implementation detail that is captured in the `tests` key. You're right - in this case we do not need an actual implemented test since the background assertions will verify this for us. So we just take an existing implemented test that has all clocks running at the right frequencies and has those assertion checks enabled and stick it in the `tests` key.